hw/ip/rv_dm/data/rv_dm_testplan.hjson

```diff
@@ -495,6 +495,50 @@
       stage: V1
       tests: []
     }
+    {
+      name: progbuf_read_write_execute
+      desc: '''
+            Verify the functionality of program buffer by writing random data on program buffer.
+
+            - Program jtag_dmi_ral.dmcontrol.haltreq to 1'b1.
+            - Verify that 'debug_req_o' is set to 1.
+            - Acknowledge haltreq by setting tl_mem_ral.halted to 0.
+            - Fill the program buffer by writing random data to jtag_dmi_ral.progbuf0-7
+              registers.
+            - Send a valid abstract command (AccessRegister) by setting postexec bit set.
+            - Read tl_mem_ral.program_buffer0-7 registers and verify that the data we wrote is
+              what we read back.
+            '''
+      stage: V1
+      tests: ["rv_dm_progbuf_read_write_execute"]
+    }
+    {
+      name: progbuf_exception
+      desc: '''
+            Verify that writing on DM progbuf registers while abstract command is executing
+            causes jtag_dmi_ral.abstractcs.cmderr to be set to 3.
+
+            - Program jtag_dmi_ral.dmcontrol.haltreq to 1'b1.
+            - Verify that 'debug_req_o' is set to 1.
+            - Acknowledge haltreq by setting tl_mem_ral.halted to 0.
+            - Send access register abstract command with postexec bit set.
+            - Write on any of progbuf registers.
+            - Write on tl_mem_ral.exception register.
```

Comment:
Why is this needed? Wouldn't we expect to have caused an error even without this line?

Reply:
So basically exception is another tl_mem_ral.exception register which is WO. This register is written by core when it is in debug mode. So here we just wanted to check that if core generate exception by writing on this register then cmderr field in abstractcs register of DM should set to 3.

Reply:
Ok, I think I understand that explanation. But now I don't understand why we're writing to the progbuf registers! Based on the initial paragraph of the description, I'd probably guess that: "If you write to a progbuf register when an abstract command is executing then you'll see an error".

If *that* is true, we shouldn't need to write to `exception` (which is why I asked the question in the first place). Your description in reply to me now makes me think that "If a core writes to `exception` then you'll see an error". That's fine, but doesn't need any discussion about `progbuf` :-)

My guess is that you're trying to say that both things need to happen to cause `abstractcs.cmderr == 3`. Would it make sense to expand the initial paragraph of the description to roughly explain what this test is trying to do?

Reply:
> Ok, I think I understand that explanation. But now I don't understand why we're writing to the progbuf registers! Based on the initial paragraph of the description, I'd probably guess that: "If you write to a progbuf register when an abstract command is executing then you'll see an error".
> 
> If _that_ is true, we shouldn't need to write to `exception` (which is why I asked the question in the first place). Your description in reply to me now makes me think that "If a core writes to `exception` then you'll see an error". That's fine, but doesn't need any discussion about `progbuf` :-)
> 
> My guess is that you're trying to say that both things need to happen to cause `abstractcs.cmderr == 3`. Would it make sense to expand the initial paragraph of the description to roughly explain what this test is trying to do?

Oh, ok i understand. i have changed the description.

Reply:
Brill, thanks!